core/modules/czar/python/appInterface.py

```diff
@@ -101,14 +113,39 @@ def submitQuery(self, query, conditions):
         else:
             lock.unlock()
             return ("error","error",a.getError())
+
+        # Remember client context for kill-operations
+        proxyName = conditions["client_dst_name"]
+        proxyThread = conditions["server_thread_id"]
+        self._clientToServerId[(proxyThread, proxyName)] = a.sessionId
+
         return (resultName, lockName, "")
 
-    def killQuery(self, query, taskId):
+    def killQuery(self, sessionId):
         """Process a kill query command (experimental).  Params: query, taskId."""
         ## Disable query killing DM-1715
-        ##a = app.KillQueryAction(query)
-        ##self._callWithThread(a.invoke)
-        return "Attempt query kill: " + query
+        a = app.KillQueryAction(sessionId)
+        self._maybeCallWithThread(a.invoke)
+        return "Attempt query kill: " + str(sessionId)
+
+    def killQueryUgly(self, killStr, clientId):
+        """Process a kill query command (experimental).  Params: killquery, clientName."""
+        #lookup sessionId using thread_id (parsed from killStr) and clientName
```

Comment:
I think killquery should probably be killQuery or killStr and clientNames should probably be clientId.